File: exam.py
#!/usr/bin/python3
"""student can only sit an exam if he has over 75% attendance"""


def attend():
    classes = int(input("How many classes were held?: "))
    attended = int(input("How many classes did you attend?: "))
    perc = (attended/classes)*100
    if perc >= 75:
        print(f"You are free to take your exams. You attended {perc}% classes")
    else:
        medical = input("Do you have medical cause for not attending\
                        enough classes? reply with 'Y' or 'N': ")
        if medical == 'Y':
            print("You can sit for the exams due to medical reasons")
        elif medical == 'N':
            print(f"You attended {perc}% classes.\
                  Not enough to sit for your exams")
        else:
            print(f"You may not sit for the exams.\
                  You attended {perc}% of classes")


if __name__ == "__main__":
    attend()

File: exam.java
import java.util.Scanner;

public class exam {
    public static void main(String[] args){
        /* a student can only sit for their exams if they attend atleast 75% of 
         * their classes or if they have medical reasons
         */
        Scanner myScanner = new Scanner(System.in);
        System.out.println("Total number of classes attended: ");
        int clas = myScanner.nextInt();
        System.out.println("Classes available? ");
        int tot = myScanner.nextInt();
        double perc = ((double) clas / tot) * 100;
        if (perc > 75) {
            System.out.printf("You are free to do the exams. You attended %.2f%% classes\n", perc);
        }
        else{
            myScanner.nextLine();
            System.out.println
            ("You did not attend enough classes. Was this because of medical reasons? Reply 'Y' or 'N': ");
            String input = myScanner.nextLine();
            char ans = input.charAt(0);
            myScanner.close();
            if (ans == 'Y'){
                System.out.printf("You attended only %.2f%% of classes. However, due to medical reasons, you can take the exams.\n", perc);
            }
            else if (ans == 'N'){
                System.out.printf("You attended only %.2f%% of classes. You will not take the exams.\n", perc);
            }
            else{
                System.out.printf("Wrong input\n");
            }
        }
    }
}
